File: Countries/bookend.vowels/bookend.vowels.py
# toy program to make a list of all the countries that begin and end with vowels

# importing the country list
countries = open("/home/evergreen/UnderDogsAssignments/Countries/countries.txt")

#turning countries.doc into an array
countries_list = []
for line in countries:
    countries_list.append(line.strip())

# Creating a list of vowels and and then going through
# countries_list to add to a countries_bookend_vowels as appropriate
vowels = ["A", "E", "I", "O", "U", "Y"]
lower_case_vowels = ["a", "e", "i", "o", "u", "y"]
bookend_vowels = []

for country in countries_list:
    # if index[0] of country is contained in vowels
    if country[0] in vowels:
        # if the last index of country is contained in vowels
        length = len(country)
        last_char = country[length -1]
        if last_char in lower_case_vowels:
            bookend_vowels.append(country)

print("All of the countries who's name begins and end with a vowel are " + str(bookend_vowels))

File: Countries/bookend.vowels/bookendVowels.js
// Program to create a list of all countries containing "United"

// Importing the fs module
const { count } = require("console");
let fs = require("fs")

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let countries = readFileLines('/home/evergreen/UnderDogsAssignments/Countries/countries.txt');

const bookendVowels = [];
const vowelSet = new Set(["A", "E", "I", "O", "U", "Y"]);

for (var i=0; i < countries.length - 1; i++) {
   let upperCaseCountry = countries[i].toUpperCase();
   if (vowelSet.has(upperCaseCountry.slice(0, 1)) && vowelSet.has(upperCaseCountry.slice(-1))) {
      bookendVowels.push(countries[i]);
   }
}

console.log(bookendVowels)

//          return vowelSet.has(x);
// && vowelSet.has(upperCaseCountry[i].slice(-1))